### PR TITLE
Remove citation entry from FAQ.

### PR DESCRIPTION
**Pull request by anntzer**
The bibtex entry is different from the one in `citing.rst` and in
`matplotlib/__init__.rst`. `citing.rst` is already linked from the
homepage (or one can search "citing"/"citation" in the search box) so I
decided against keeping it in the FAQ.

While we're at it, fix the citation in citing.rst (it's not Hunter et al
-- jdh was the sole author).

## PR Checklist

- [ ] Has Pytest style unit tests
- [ ] Code is [Flake 8](http://flake8.pycqa.org/en/latest/) compliant
- [ ] New features are documented, with examples if plot related
- [ ] Documentation is sphinx and numpydoc compliant
- [ ] Added an entry to doc/users/next_whats_new/ if major new feature (follow instructions in README.rst there)
- [ ] Documented in doc/api/api_changes.rst if API changed in a backward-incompatible way

### CODE REVIEW

File: doc/citing.rst [hunk 1]
```diff
@@ -3,23 +3,25 @@
 Citing Matplotlib
 =================
 
-If Matplotlib contributes to a project that leads to a scientific
-publication, please acknowledge this fact by citing `Hunter et al (2007)
-<https://doi.org/10.1109/MCSE.2007.55>`_ using this ready-made BibTeX entry:
+If Matplotlib contributes to a project that leads to a scientific publication,
+please acknowledge this fact by citing `J. D. Hunter, "Matplotlib: A 2D
+Graphics Environment", Computing in Science & Engineering, vol. 9, no. 3,
+pp. 90-95, 2007 <https://doi.org/10.1109/MCSE.2007.55>`_.
```
Comment (anntzer) [inline doc/citing.rst]: (This is IEEE style, which is, well, consistent with the journal this was published in...)

File: doc/citing.rst [hunk 2]
```diff
@@ -3,23 +3,25 @@
...
+
+for example by using this ready-made BibTeX entry:
```
Comment (story645) [inline doc/citing.rst]: all out nitpick, but I don't think you need this line

Comment (anntzer) [reply doc/citing.rst]: I don't really care, but sure.

**Comment by timhoffm** [comment]
Semi-OT: IMHO the citing section on the main page could also go. Citing is probably not one of the most common things that users are looking for. OTOH it would be good to add citing.rst to the contents, somewhere near Users Guide -> License or Credits.

**Comment by anntzer** [comment]
You get to push whatever fix you want :)

**Comment by timhoffm** [comment]
> You get to push whatever fix you want :)

On the road! So, can just post nagging comments for now. 😛 

> Edit: Actually, nah, it doesn't cost much and kind of makes sense to leave it there, next to donations and all.
I’ve some overall concerns with the main page! But that’s for another discussion (#13382).

@tacaswell Agree. 😄

**Pull request merged**